backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ElasticSearchHealthCheck.java

```diff
@@ -54,30 +55,32 @@ public ComponentName componentName() {
     }
 
     @Override
-    public Result check() {
-        String[] indices = indexNames.stream()
-            .map(IndexName::getValue)
-            .toArray(String[]::new);
-        ClusterHealthRequest request = Requests.clusterHealthRequest(indices);
+    public Mono<Result> check() {
+        return createRequestFromIndexNames()
+            .flatMap(request -> Mono.fromCallable(() ->
+                client.cluster().health(request, RequestOptions.DEFAULT)))
+            .flatMap(this::toHealthCheckResult)
```

Comment:
use only `map`.

Reply:
I can agree with using map for toHealthCheckResult, but the health() function will become more verbose, like:

`.map(Throwing.<ClusterHealthRequest, ClusterHealthResponse>function(request ->
                client.cluster().health(request, RequestOptions.DEFAULT)).sneakyThrow())`

Is there any good reason behind the use of map in this case, or it is purely cosmetic?

Reply:
I agree with `flatMap` it looks more clean.

Reply:
Semantically it's very different `map` guaranties the element number, while `flatMap` means that the number of elements can change and/or the way the the transformation is done requires a special operation, which confuses the reader.

Reply:
alright! I was reading the code too fast.